source/code/plugin/in_kube_podinventory.rb

```diff
@@ -277,6 +278,13 @@ def parse_and_emit_records(podInventory, serviceList)
             record["ControllerName"] = items["metadata"]["ownerReferences"][0]["name"]
             if telemetryFlush == true
               controllerSet.add(record["ControllerKind"] + record["ControllerName"])
+              #Adding controller kind to telemetry ro information about customer workload
+              if (controllerData[record["ControllerKind"]].nil?)
```

Comment:
check for not nil

Reply:
There's a check for ownerreferences that i forgot to mention. Its hidden. So resolving this.